lms/djangoapps/commerce/tests.py

```diff
@@ -0,0 +1,247 @@
+        """
+        If the course has a SKU, the view should get authorization from the E-Commerce API before enrolling
+        the user in the course. If authorization is approved, the user should be redirected to the user dashboard.
+        """
+
+        # Set user's active flag
+        self.user.is_active = user_is_active
+        self.user.save()  # pylint: disable=no-member
+
+        def request_callback(_method, _uri, headers):
+            """ Mock the E-Commerce API's call to the enrollment API. """
+            add_enrollment(self.user.username, unicode(self.course.id), 'honor')
+            return 200, headers, ECOMMERCE_API_SUCCESSFUL_BODY
+
+        self._mock_ecommerce_api(body=request_callback)
+        response = self._post_to_view()
+
+        # Validate the response content
+        msg = 'Order {} was completed.'.format(ORDER_NUMBER)
+        self.assertResponseMessage(response, msg)
+
+        self.assertTrue(CourseEnrollment.is_enrolled(self.user, self.course.id))
+
+        # Verify the correct information was passed to the E-Commerce API
+        request = httpretty.last_request()
+        sku = CourseMode.objects.filter(course_id=self.course.id, mode_slug='honor', sku__isnull=False)[0].sku
+        self.assertEqual(request.body, 'sku={}'.format(sku))
+        self.assertEqual(request.headers['Content-Type'], 'application/json')
+
+        # Verify the JWT is correct
+        expected_jwt = jwt.encode({'username': self.user.username, 'email': self.user.email},
+                                  ECOMMERCE_API_SIGNING_KEY)
+        self.assertEqual(request.headers['Authorization'], 'JWT {}'.format(expected_jwt))
+
+    @httpretty.activate
+    def test_order_not_complete(self):
+        self._mock_ecommerce_api(body=json.dumps({'status': OrderStatus.OPEN, 'number': ORDER_NUMBER}))
+        response = self._post_to_view()
+        self.assertEqual(response.status_code, 202)
+        msg = u'Order {} was created, but is not yet complete. User was enrolled.'.format(ORDER_NUMBER)
+        self.assertResponseMessage(response, msg)
+
+        # TODO Eventually we should NOT be enrolling users directly from this view.
+        self.assertTrue(CourseEnrollment.is_enrolled(self.user, self.course.id))
+
+    @httpretty.activate
+    def test_course_without_sku(self):
+        """
+        If the course does NOT have a SKU, the user should be enrolled in the course (under the honor mode) and
+        redirected to the user dashboard.
+        """
+        # Remove SKU from all course modes
+        for course_mode in CourseMode.objects.filter(course_id=self.course.id):
+            course_mode.sku = None
+            course_mode.save()
+
+        # Place an order
+        self._mock_ecommerce_api()
+        response = self._post_to_view()
+
+        # Validate the response content
+        self.assertEqual(response.status_code, 200)
+        msg = 'The honor mode for {} does not have a SKU. Enrolling {} directly.'.format(
```

Comment:
If you were to make this message and others like it constants in the views, you could import them here for use when performing these assertions. That way, if these strings change, the tests won't need to be updated.